regular_expressions_basics/regular_expressions_basics.md

```diff
@@ -0,0 +1,338 @@
+version:  1.0.0
+current_version_description: Initial version
+module_type: standard
+docs_version: 1.0.0
+language: en
+narrator: UK English Female
+mode: Textbook
+previous_sequential_module: demystifying_regular_expressions
+
+title: Regular Expressions Basics
+
+comment:  Begin to use regular expressions, or regex, for simple pattern matching.
+
+long_description: Regular expressions, or regex, are a way to specify patterns (such as the pattern that defines a valid email address, medical record number, or credit card number).  Learn to compose basic regular expressions in order to find and use important data.
+
+estimated_time_in_minutes: 60
+
+@pre_reqs
+Learners should have some knowledge about patterns in biomedical data and understand the utility of regular expressions (regex).  For an introduction to these concepts, consider the [Demystifying Regular Expressions](https://liascript.github.io/course/?https://raw.githubusercontent.com/arcus/education_modules/main/demystifying_regular_expressions/demystifying_regular_expressions.md#1) module.
+@end
+
+@learning_objectives  
+After completion of this module, learners will be able to:
+
+- Define a simple alphanumeric pattern in regex notation
+- List common ranges and character groups in regex
+- Quantify characters appearing optionally, once, or multiple times in regex
+- Test regular expressions in a browser
+@end
+
+@version_history 
+No previous versions.
+@end
+
+import: https://raw.githubusercontent.com/arcus/education_modules/main/_module_templates/macros.md
+-->
+
+# Regular Expressions Basics
+
+@overview
+
+## Regular Expression Refresher
+
+As a reminder, the word *regular* (like *regulate*) comes from the Latin root for "rule": *regula*.  A **regular expression** (also known as "regex") is a specific way to **express** a **rule** for a pattern, such as "three or more upper case letters followed by exactly two digits, the first of which cannot be a 0, followed by a decimal point and two more digits".  Regular expressions can look like an odd combination of characters.  
+
+One example? `[A-Z]{3,}[1-9]\d\.\d{2}` represents the "three or more upper case letters" pattern described above.
+
+Regular expressions are useful to biomedical researchers because they can allow you to find, replace, or extract text that matches patterns you define.  These text patterns might appear in your data (for example, clinical notes), in metadata (such as column names), in your data analysis scripts (for example, where you define hexadecimal codes for color), and in file names.  
+
+A good place to practice regular expressions is an online regex testing website.  We like "Regex 101", at https://www.regex101.com, but you can also search online for "regex checker" to find other, similar websites.  Regex 101 allows you to enter your regular expression as well as some sample text that you want to apply the regular expression to.  We will ask you to use this website to gain practice during this module!
```

Comment:
This paragraph doesn't quite fit into this module, now that the regex101.com stuff is in a different module. I think it can just come out.

Reply:
Yes, but what if people don't do each module?  Ideally each module should be freestanding, so a bit of "hey this is the one we suggest" makes sense in each module, I think?

Reply:
Ah, good point! I think I was mainly thrown by the last sentence promising that we will have them practice using online checkers. How is this:

```suggestion
A good place to practice regular expressions is an online regex testing website.  We like "Regex 101", at https://www.regex101.com, but you can also search online for "regex checker" to find other, similar websites.  Regex 101 allows you to enter your regular expression as well as some sample text that you want to apply the regular expression to.  You can use this website to gain practice with any of the rules you learn in this module.

The [Regex 101 section](link) in the Demystifying Regular Expressions module shows and example of how to use the website.
```